Added docs on how to configure the reverse proxy for a subfolder & Documented the usage of X-Forwared-Prefix (#19128)
symfony/symfony-docs: deployment/proxies.rst

ranges of any additional proxy (e.g. `CloudFront IP ranges`_) to the array of
trusted proxies.

Reverse proxy in a subpath / subfolder
--------------------------------------

> If you configure the reverse proxy in such a way that the Symfony application is

Reviewer: if i understand correctly, this is about the proxy routing to symfony for `/subpath`, but symfony itself running on `/`? in a way this is the only thing that makes sense for this documentation, nonetheless i would make that more explicit. you could start with the example and explicitly explain that the symfony application runs at `/`, and give the example of what absolute paths/urls should be.

Reviewer 2: @jbtronics if possible, please add a phrase or two as a short introduction to the exact problem that this solves, as asked by David. Thanks!

ranges of any additional proxy (e.g. `CloudFront IP ranges`_) to the array of
trusted proxies.

Reverse proxy in a subpath / subfolder
--------------------------------------

If you configure the reverse proxy in such a way that the Symfony application is
mapped in a subpath / subfolder instead of the root path of the web host, you will need to
pass the prefix to Symfony by setting the ``X-Forwarded-Prefix`` header.
This is required for Symfony to generate the correct URLs. The header can normally be configured

> in your reverse proxy configuration. Be sure that ``X-Forwared-Prefix`` is configured as trusted header, if

Reviewer: ```suggestion
in your reverse proxy configuration. Configure ``X-Forwared-Prefix`` as trusted header to be able to use this feature.
```

mapped in a subpath / subfolder instead of the root path of the web host, you will need to
pass the prefix to Symfony by setting the ``X-Forwarded-Prefix`` header.
This is required for Symfony to generate the correct URLs. The header can normally be configured
in your reverse proxy configuration. Configure ``X-Forwared-Prefix`` as trusted header to be able to use this feature.

> you want to use this feature.

Reviewer: ```suggestion
```
